test/ui/browser/actions/test-set-user-typed-location.js

```diff
@@ -52,4 +54,28 @@ describe('Action - SET_USER_TYPED_LOCATION', () => {
     expect(getState().pages.get(1).userTyped).toEqual(null);
     expect(getState().pages.get(2).userTyped).toEqual(null);
   });
+
+  it('Should send a message to the main process', function(done) {
+    const { dispatch } = this;
+
+    ipcMainMock.on('profile-command', handleIpc);
+
+    dispatch(actions.setUserTypedLocation({
+      pageIndex: 1,
+      text: 'Bar',
+    }));
+
+    function handleIpc(e, ...args) {
+      // Filter out any mock ipc calls that are not yet guaranteed to have
+      // completed
+      if (args[0].type !== profileConstants.SET_USER_TYPED_LOCATION ||
```

Comment:
Let's set up a more declarative way to specify what packets are expected.  Feel free to do in follow up though